apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCardButtons.jsx

```diff
@@ -142,11 +171,14 @@ export default class ActivitySectionCardButtons extends Component {
           isOpen={this.state.addResourceOpen}
           handleConfirm={this.handleCloseAddResource}
         />
-        <EditTipDialog
-          isOpen={this.state.addTipOpen}
-          handleConfirm={this.handleCloseAddTip}
-          tip={this.state.tipToEdit}
-        />
+        {this.state.addTipOpen && (
+          <EditTipDialog
+            isOpen={this.state.addTipOpen}
```

Comment:
Can you pick one of these two ways to close / hide the dialog, and remove the other? As it is, the EditTipDialog looks like it would never be rendered with `isOpen={false}`.  You're welcome to pick whichever seems easiest. 🙂

Reply:
So isOpen is required by the BaseDialog component used in EditTipDialog and it does not prevent the component from trying to render. This became an issue when the tip that you are trying to edit has not been set yet. So the outer check prevents the EditTipDialog from rendering. Thoughts?

Reply:
interesting. In that case, I would question why we are able to get into a state where the dialog is told to be open without having set the tip that you are trying to edit. is this in the add or edit case? in both cases, `handleEditTip` and `handleOpenAddTip`, you set `tipToEdit` and `addTipOpen` in the same call to `setState`, so I would expect these state objects to both be updated in the same render call.

If nothing obvious jumps out, I would suggest hard-coding `isOpen={true}` here as an alternative, or just adding a short comment explaining why both are needed.

Reply:
It happens on the initial render of the Lesson Edit page when I haven't opened the dialog at all. I tested printing out isOpen and Tip in EditTipDialog and they are correct (false and null when first rendered) but it was still giving an error. I went with the isOpen={true} approach